### ISSUE TITLE
AttributeManager: add support to calculate bounding box

### ISSUE THREAD

**Issue by 1chandu** [visgl/deck.gl#3828]
#### Proposed feature

AttributeManager parses `data` array to build attributes, in some uses cases (like GPUGridLayer), it needs to know the bounding box of the data like (lng/lat positions). 
Add a flag `needBounds` to attribute definition object, when set `AttributManager` will clacuate bounds and layers can acces it as`attribute.bounds`.

```
 attributeManager.add({
      positions: {size: 3, accessor: 'getPosition', type: GL.DOUBLE, fp64: false, needBounds: true},
     ...
    });

...

 const {min, max} = attributeManager.getAttributes().positions.bounds;
...
```

#### To Do List 
  - [x] Add label and assign to milestone 
  - [ ] Coding
  - [ ] Doc update
  - [ ] What’s new update 
  - [ ] Test

**Comment by ibgreen** [comment]
+1 

Though given our direction, a concern is how to calculate bounds for `GPUTable`s... 
- Are you able to use GPGPU techniques to aggregate max and min values in a small texture? 
- Or do we need to equip the GPUTable columns with a bounds metadata?
- Or only support this for JS arrays?

**Comment by 1chandu** [comment]
> Are you able to use GPGPU techniques to aggregate max and min values in a small texture?

Yes, this technique is currently being used in our aggregation layers, provided rendering to float textures is supported.

This issue is mostly applicable to when data prop is array of JS objects, given we are already parsing each value to build JS array of an attribute, calculating bounds should be cheap.

**Comment by ibgreen** [comment]
We have already implemented a culling math library. If we add a `boundingBox` prop (or `state.boundingBox` field) on each layer, we could automatically cull anything outside the current view frustum.

The boudingBox could be autocalculated or supplied by app.

**Comment by 1chandu** [comment]
Moving to 8.1
